src/providers/irc/AbstractIrcServer.cpp

```diff
@@ -279,12 +279,20 @@ void AbstractIrcServer::onReadConnected(IrcConnection *connection)
 
     std::lock_guard lock(this->channelMutex);
 
+    // Apply a leaky bucket rate limitting to JOIN mesasges
+    this->bucket_ =
+        std::make_unique<RatelimitBucket>(18, 10500, [&](QString message) {
+            qCDebug(chatterinoIrc) << "joining" << message;
+            connection->sendRaw("JOIN #" + message);
+        });
+
     // join channels
     for (auto &&weak : this->channels)
     {
         if (auto channel = weak.lock())
         {
-            connection->sendRaw("JOIN #" + channel->getName());
+            qCDebug(chatterinoIrc) << "sending to bucket" << channel->getName();
```

Comment:
Phrasing changed to `joining` as that's what will happen immediately as the callback is called

Reply:
Nevermind my last comment - I read the wrote qCDebug call. Fixed in a new commit.

Reply:
These debug prints were only added to ensure my logic works while preparing first commit to this PR and I was planning on removing it after code reviewers agree it works. Maybe it can stay though.